### ISSUE TITLE
Document in plugin writer guide tasks are not safe to wait on other tasks

### ISSUE THREAD

**Issue by pulpbot** [pulp/pulpcore#1896]
Author: @bmbouter (bmbouter)

Redmine Issue: 6773, https://pulp.plan.io/issues/6773

---

## Background

Pulp architecturally cannot have a task that waits on another task, otherwise on systems where only 1 pulp worker is running Pulp will deadlock. Pulp workers only work on one task at a time, so here is the deadlock example:

1) Task foo starts running
2) Task foo dispatches task bar
3) Task bar never starts because Task foo never finishes and there is only one worker
4) despair

## Solution

Document a new section named "tasks waiting on tasks" in the plugin writer's guide [Tasks section here](https://docs.pulpproject.org/plugins/plugin-writer/concepts/index.html#tasks). It should document this.
